src/components/Visualization.tsx

```diff
@@ -60,6 +60,10 @@ const Visualization = () => {
     const step = stepsService.findStepWithUUID(selectedStep.UUID);
     if (step) {
       setSelectedStep(step);
+    } else {
+      setSelectedStep({ maxBranches: 0, minBranches: 0, name: '', type: '', UUID: '' });
+      visualizationStore.setSelectedStepUuid('');
+      if (isPanelExpanded) setIsPanelExpanded(false);
```

Comment:
Should we check whether it's closed or not? I think that this update will happen inside of an update batch, so I'm wondering if we can simply set it to `false` and call it a day :smile: 

```suggestion
      setIsPanelExpanded(false);
```